devicetypes/smartthings/tile-ux/tile-multiattribute-generic.src/tile-multiattribute-generic.groovy

```diff
@@ -0,0 +1,118 @@
+/**
+ *  Copyright 2016 SmartThings, Inc.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License. You may obtain a copy of the License at:
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software distributed under the License is distributed
+ *  on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License
+ *  for the specific language governing permissions and limitations under the License.
+ *
+ */
+metadata {
+	definition (name: "genericDeviceTile", namespace: "smartthings/tile-ux", author: "SmartThings") {
+		capability "Actuator"
+		capability "Switch"
+		capability "Switch Level"
+
+		command "levelUp"
+		command "levelDown"
+		command "randomizeLevel"
+	}
+
+	tiles(scale: 2) {
+		multiAttributeTile(name:"basicTile", type:"generic", width:6, height:4) {
+			tileAttribute("device.switch", key: "PRIMARY_CONTROL") {
+				attributeState "on", label:'${name}', action:"switch.off", icon:"st.switches.switch.on", backgroundColor:"#79b821", nextState:"turningOff"
+				attributeState "off", label:'${name}', action:"switch.on", icon:"st.switches.switch.off", backgroundColor:"#ffffff", nextState:"turningOn"
+				attributeState "turningOn", label:'${name}', action:"switch.off", icon:"st.switches.switch.on", backgroundColor:"#79b821", nextState:"turningOff"
+				attributeState "turningOff", label:'${name}', action:"switch.on", icon:"st.switches.switch.off", backgroundColor:"#ffffff", nextState:"turningOn"
+			}
+		}
+		multiAttributeTile(name:"sliderTile", type:"generic", width:6, height:4) {
+			tileAttribute("device.switch", key: "PRIMARY_CONTROL") {
+				attributeState "on", label:'${name}', backgroundColor:"#79b821", nextState:"turningOff"
+				attributeState "off", label:'${name}', backgroundColor:"#ffffff", nextState:"turningOn"
+				attributeState "turningOn", label:'${name}', backgroundColor:"#79b821", nextState:"turningOff"
+				attributeState "turningOff", label:'${name}', backgroundColor:"#ffffff", nextState:"turningOn"
+			}
+			tileAttribute("device.level", key: "SECONDARY_CONTROL") {
+				attributeState "default", icon: 'st.Weather.weather1', action:"randomizeLevel"
```

Comment:
This icon appears very small (due to the large amounts of whitespace surrounding the image) and non-white. We should probably have another icon here.